iop/cdvd/cdfs/src/cdfs_iop.c

```diff
@@ -0,0 +1,1263 @@
+            tocEntryPointer = (struct DirTocEntry *)((u8 *)tocEntryPointer + tocEntryPointer->length);
+
+        dir_entry = 0;
+
+        while (1) {
+#ifdef DEBUG
+            printf("cdfs_getDir - inside while-loop\n\n");
+#endif
+
+            // parse the current cache block
+            for (; tocEntryPointer < (struct DirTocEntry *)(cacheInfoDir.cache + (cacheInfoDir.cache_size * 2048)); tocEntryPointer = (struct DirTocEntry *)((u8 *)tocEntryPointer + tocEntryPointer->length)) {
+                if (tocEntryPointer->length == 0) {
+                    // if we have a toc entry length of zero,
+                    // then we've either reached the end of the sector, or the end of the dir
+                    // so point to next sector (if there is one - will be checked by next condition)
+
+                    tocEntryPointer = (struct DirTocEntry *)(cacheInfoDir.cache + (((((char *)tocEntryPointer - cacheInfoDir.cache) / 2048) + 1) * 2048));
+                }
+
+                if (tocEntryPointer >= (struct DirTocEntry *)(cacheInfoDir.cache + (cacheInfoDir.cache_size * 2048))) {
+                    // we've reached the end of the current cache block (which may be end of entire dir
+                    // so just break the loop
+                    break;
+                }
+
+                // Check if the current entry is a dir or a file
+                if (tocEntryPointer->fileProperties & 0x02) {
+#ifdef DEBUG
+                    printf("We found a dir, and we want all dirs\n\n");
+#endif
+
+                    // wait for any previous DMA to complete
+                    // before over-writing localTocEntry
+                    while (sceSifDmaStat(dmaID) >= 0) {}
+                    copyToTocEntry(&localTocEntry, tocEntryPointer);
+
+                    if (dir_entry == 0) {
+                        if (cacheInfoDir.path_depth != 0) {
+#ifdef DEBUG
+                            printf("It's the first directory entry, so name it '..'\n\n");
+#endif
+                            strcpy(localTocEntry.filename, "..");
+                        }
+                    }
+
+                    // DMA localTocEntry to the address specified by tocEntry[matched_entries]
+
+                    // setup the dma struct
+                    dmaStruct.src = &localTocEntry;
+                    dmaStruct.dest = &tocEntry[matched_entries];
+                    tocEntry[matched_entries] = localTocEntry;
+                    dmaStruct.size = sizeof(struct TocEntry);
+                    dmaStruct.attr = 0;
+
+                    // Do the DMA transfer
+                    CpuSuspendIntr(&intStatus);
+
+                    dmaID = sceSifSetDma(&dmaStruct, 1);
+
+                    CpuResumeIntr(intStatus);
+
+                    matched_entries++;
+                } else  // it must be a file
+                {
+#ifdef DEBUG
+                    printf("We found a file, but we dont want files (at least not yet)\n\n");
+#endif
```

Comment:
These 3 lines are in many places. The code would be more readable when using for instance DPRINTF, defined once at the top of the file.

Reply:
Done!